### PR TITLE
Login page

### PR DESCRIPTION
**Pull request by Drapegnik**
## made
* Add login form component with logic
* Add auth stories (`FormField`)
* Add `copy-static.sh` script for copy images, styles and fonts in `static` for develop mode
* Add login page
* Add form client side validation
* Add redirect on success
* Extract text constants to file (dictionary)
* Add [`prettier`](prettier.io) support, fix some [`eslint`](https://eslint.org/) rules

## to review
* [x] Please test script (should run automatically after `npm i` or `npm run update-styles`)
* [x] Please check login page logic (validation, requests on server and etc.)

## demo
screen shot 2017-11-26 at 17 15 55

screen shot 2017-11-26 at 17 22 21

### CODE REVIEW

**Comment by Fusionqq** [comment]
Maybe make sense doing login button smaller and with non-transparent background?

**Review by Fusionqq — DISMISSED** [review]

**Comment by UladBohdan** [comment]
Not sure what to test here.
But infinite pending looks weird: it spins too fast, and not clear why we need two spinners (or even three when it's a signup mode) - what if we add one on "Паехалі" button?

And remove letter "д" in "Пацвердзіце" word plz.

**Review by UladBohdan — CHANGES_REQUESTED** [review]
Not sure what to test here.
But infinite pending looks weird: it spins too fast, and not clear why we need two spinners (or even three when it's a signup mode) - what if we add one on "Паехалі" button?

And remove letter "д" in "Пацвердзіце" word plz.

**Comment by Drapegnik** [comment]
@UladBohdan, redirect issues to `@babajka/markup` team.

File: components/auth/LoginForm.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        >
+          <input className="input password-input" type="password" placeholder="Пароль" />
+        </FormControl>
+
+        {signUpMode && (
+          <FormControl
+            error={errors.passwordAgain}
+            success={successStatus.passwordAgain}
+          >
+            <input className="input password-input" type="password" placeholder="Падцвердзіце пароль" />
```
Comment (UladBohdan) [inline components/auth/LoginForm.js]: Even if it's fixed in markup, here we still have "д" mistype. Should be fixed both here and in markup.

**Review by UladBohdan — CHANGES_REQUESTED** [review]
But you're manually reapplying all html-markup changes. Redirecting to markup team won't help.

**Comment by Drapegnik** [comment]
[login page frontend](https://trello.com/c/VtDqIPmE/66-login-page-frontend)

**Review by UladBohdan — DISMISSED** [review]

**Comment by UladBohdan** [comment]
Not sure why package-lock.json got updated.

**Comment by UladBohdan** [comment]
I've created a new account and was able to log into it. Awesome!

Some minor questions/suggestions:
1. what's the plan about "forgot password" button?
2. can you change the tab order so that on login after I've filled in the email field and press tab - I'm on the password field? that would be much more convenient. Now there's this intermediate "forgot password" position.
3. it doesn't save the session, is that fine for now?
4. what about reseting all warning once I switch between login and signup? It's a bit weird: I put the cursor into the "email" field, type nothing, escape the cursor from the field - the warning "bad email" appears - than I switch into the signup mode - but the warning it still present.

And it seems to me there's a rule: if user failed to login, we NEVER indicate if email is incorrect or passport doesn't match. That gives too much information. We must return "failed to login" error in any case. That's a fix for both backend and frontend.

**Comment by Drapegnik** [comment]
@UladBohdan:
1. hide `forgot password` btn for now, not urgent feature
2. in future can be fixed with [`tabindex`](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/tabindex)
3. there is [`task`](https://trello.com/c/HGRitZDA/76-auth-cookies-bugs) for it
4. fixed it: clear set `touched: false` for empty fields when mode switched
5. `failed to login` should be fixed on backend

**Review by UladBohdan — APPROVED** [review]

**Pull request merged**